From a1f6747606484c47581172b36aa70aa73a13ab6a Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Mon, 6 Jan 2025 16:30:26 +0900
Subject: [PATCH] mkosi: replace deprecated settings and command with new ones

Follow-up for c008b703701e676a88e16de4c867b9fc4818070d.

These settings and command were deprecated by the following commit.
https://github.com/systemd/mkosi/commit/ebcd2efa03af5c04de91434634f2e4a956f7e53c

(cherry picked from commit d5cdcba928823faf5ddcca267e3866ab5d5fc729)
---
 docs/HACKING.md | 6 +++---
 mkosi.conf      | 8 ++++----
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/docs/HACKING.md b/docs/HACKING.md
index de3f66dd55..b6aeff11c3 100644
--- a/docs/HACKING.md
+++ b/docs/HACKING.md
@@ -51,7 +51,7 @@ To build and boot an OS image with the latest systemd installed:
 $ mkosi -f genkey                               # Generate signing keys once.
 $ mkosi -f sandbox meson compile -C build mkosi # (re-)build the OS image
 $ sudo mkosi boot                               # Boot the image with systemd-nspawn.
-$ mkosi qemu                                    # Boot the image with qemu.
+$ mkosi vm                                      # Boot the image with qemu.
 ```
 
 Putting this all together, here's a series of commands for preparing a patch for
@@ -67,7 +67,7 @@ $ $EDITOR src/core/main.c                       # or wherever you'd like to make
 $ mkosi -f sandbox meson setup build            # Set up meson
 $ mkosi -f genkey                               # Generate signing keys once.
 $ mkosi -f sandbox meson compile -C build mkosi # (re-)build the test image
-$ mkosi qemu                                    # Boot the image in qemu
+$ mkosi vm                                      # Boot the image in qemu
 $ git add -p                                    # interactively put together your patch
 $ git commit                                    # commit it
 $ git push -u <REMOTE>                          # where REMOTE is your "fork" on GitHub
@@ -275,7 +275,7 @@ To simplify debugging systemd when testing changes using mkosi, we're going to s
 
 To allow VSCode's debugger to attach to systemd running in a mkosi image,
 we have to make sure it can access the virtual machine spawned by mkosi where systemd is running.
-After booting the image with `mkosi qemu`,
+After booting the image with `mkosi vm`,
 you should now be able to connect to it by running `mkosi ssh` from the same directory in another terminal window.
 
 Now we need to configure VSCode.
diff --git a/mkosi.conf b/mkosi.conf
index ab8a6a1372..2517aba518 100644
--- a/mkosi.conf
+++ b/mkosi.conf
@@ -140,10 +140,10 @@ Credentials=
         tty.console.login.noauth=yes
 RuntimeBuildSources=yes
 RuntimeScratch=no
-QemuSmp=2
-QemuSwtpm=yes
-QemuVsock=yes
-QemuKvm=yes
+CPUs=2
+TPM=yes
+VSock=yes
+KVM=yes
 
 [Include]
 Include=%D/mkosi.sanitizers
